src/modules/wreck/wrexecd.c

```diff
@@ -2382,24 +2383,24 @@ int main (int ac, char **av)
     if (!prog_ctx_getopt (ctx, "no-pmi-server") && prog_ctx_initialize_pmi (ctx) < 0)
         wlog_fatal (ctx, 1, "failed to initialize pmi-server");
 
-    if (exec_commands (ctx) == 0) {
+    exec_rc = exec_commands (ctx);
 
-        if (flux_reactor_run (flux_get_reactor (ctx->flux), 0) < 0)
-            wlog_err (ctx, "flux_reactor_run: %s", flux_strerror (errno));
-
-        rexec_state_change (ctx, "complete");
-        wlog_msg (ctx, "job complete. exiting...");
-
-        lua_stack_call (ctx->lua_stack, "rexecd_exit");
-    }
+    if (!exec_rc && flux_reactor_run (flux_get_reactor (ctx->flux), 0) < 0)
```

Comment:
Yes, it is technically valid. However, In cases like this where `rc < 0` means failure and `rc == 0` means success, I find use of `if (rc)` and `if (!rc)` a possible source for confusion because my brain tends to promote the variable to a boolean --  moreso when the test is split from the call site and I may have lost a modicum of context.. In this specific case the the line reads as "if *not* exec_rc" which is much more confusing to me than "if exec_rc is zero".

However, I'm willing to admit this could be my personal hangup and it doesn't matter enough to block merge of this PR.